raiden/network/proxies/token_network.py

```diff
@@ -812,96 +846,168 @@ def _set_total_deposit(
+                    detail_for=self.node_address,
+                    partner=partner,
+                    block_identifier=failed_at_blockhash,
+                )
+
+                channel_state = self._get_channel_state(
+                    participant1=self.node_address,
+                    participant2=partner,
+                    block_identifier=failed_at_blockhash,
+                    channel_identifier=channel_identifier,
+                )
+                # Check if deposit is being made on a nonexistent channel
+                if channel_state != ChannelState.OPENED:
+                    msg = "Deposit failed as the channel was not open"
+                    raise RaidenUnrecoverableError(msg)
+
+                deposit_amount = total_deposit - our_details.deposit
+
+                total_channel_deposit = total_deposit + partner_details.deposit
+                if total_channel_deposit >= UINT256_MAX:
+                    raise RaidenUnrecoverableError("Deposit overflow")
+
+                total_deposit_done = our_details.deposit >= total_deposit
+                if total_deposit_done:
+                    raise DepositMismatch("Requested total deposit was already performed")
+
+                channel_participant_deposit_limit = self.channel_participant_deposit_limit
+                if total_deposit > channel_participant_deposit_limit:
+                    msg = (
+                        f"Deposit of {total_deposit} is larger than the "
+                        f"channel participant deposit limit"
+                    )
+                    raise RaidenUnrecoverableError(msg)
+
+                token_network_deposit_limit = self.token_network_deposit_limit
+                network_balance = self.token.balance_of(
+                    address=Address(self.address), block_identifier=receipt["blockHash"]
+                )
+
+                if network_balance + deposit_amount > token_network_deposit_limit:
+                    msg = (
+                        f"Deposit of {deposit_amount} will have "
+                        f"exceeded the token network deposit limit."
+                    )
+                    raise RaidenUnrecoverableError(msg)
+
+                allowance = self.token.allowance(
+                    owner=self.node_address,
+                    spender=Address(self.address),
+                    block_identifier=failed_at_blockhash,
+                )
+                has_sufficient_balance = (
+                    self.token.balance_of(self.node_address, given_block_identifier)
+                    < amount_to_deposit
+                )
+                if allowance < amount_to_deposit:
+                    msg = (
+                        f"The allowance of the {amount_to_deposit} deposit changed. "
+                        f"Check concurrent deposits "
+                        f"for the same token network but different proxies."
+                    )
```

Comment:
This is missing the raise, this should be an unrecoverable exception.

Reply:
Actually, I was wrong here.

My initial rationale was that Raiden does not support concurrent usage of an Ethereum accounts. For the allowance to be small either:

- It was set to a lower value (by this or another transaction)
- It was used by another transaction

Both of the above scenarios would be bugs in Raiden (assuming control of the account). Or concurrent usage of the account (not supported).

However, I forgot to consider the existence of bad tokens. This could be an attack vector.

Reply:
The exception here should be recoverable